src/app/users-page/author/edit-authors/edit-authors.component.spec.ts

```diff
@@ -1,20 +1,57 @@
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
-
 import { EditAuthorsComponent } from './edit-authors.component';
+import { ReactiveFormsModule, FormsModule } from '@angular/forms';
+import { HttpClientModule } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { Author } from 'src/app/models/author';
+import { BrowserModule } from '@angular/platform-browser';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { AuthorService } from '../../../services/author.service';
+
+class MockService {
+  updateAnAuthor(): Observable<Author> {
+    return {} as Observable<Author>;
+  }
+}
 
 describe('EditAuthorsComponent', () => {
   let component: EditAuthorsComponent;
   let fixture: ComponentFixture<EditAuthorsComponent>;
+  let spy: any;
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
       declarations: [EditAuthorsComponent],
+      imports: [
+        BrowserModule,
+        ReactiveFormsModule,
+        FormsModule,
+        HttpClientTestingModule,
+      ],
+      providers: [
+        {
+          AuthorService,
+          useValue: MockService,
+        },
+      ],
     }).compileComponents();
+
+    fixture = TestBed.createComponent(EditAuthorsComponent);
+    component = fixture.componentInstance;
+    component.ngOnInit();
```

Comment:
The **ngOnInit** method of this component makes a call to **getAllAuthors** which in turn fetches authors from the backend. Consider also adding this method to your **mockService** above to prevent this network call.